modules/mediawiki/files/bin/deploy-mediawiki.py

```diff
@@ -4,20 +4,105 @@
+        'wikiurl': 'beta.betaheze.org',
+        'servers': [],
+        'canary': 'test101',
+    },
+    'prod': {
+        'wikidbname': 'testwiki',  # don't use loginwiki anymore - we want this to be an experimental wiki
+        'wikiurl': 'publictestwiki.com',
+        'servers': ['mw101', 'mw102', 'mw111', 'mw112', 'mw121', 'mw122'],
+        'canary': 'mwtask111',
+    }
+}
+HOSTNAME = socket.gethostname()
 
+def get_environment_info():
+    if HOSTNAME.split('.')[0].startswith('test'):
+        return ENVIRONMENTS['beta']
+    return ENVIRONMENTS['prod']
 
-def check_up(server):
+
+def get_server_list(envinfo, servers):
+    if servers in ('all', 'scsvg'):
+        slist = envinfo['servers']
+        return slist.append(envinfo['canary'])
+    if servers == 'skip':
+        return [envinfo['canary']]
+    return servers.split(',')
+    
+    
+
+
+def run_command(cmd):
+    start = time.time()
+    print(f'Execute: {cmd}')
+    ec = os.system(cmd)
+    print(f'Completed ({ec}) in {str(int(time.time() - start))}s!')
+    return ec
+
+
+def non_zero_code(ec, exit=True):
+    for code in ec:
+        if code != 0:
+            if exit:
+                print('Exiting due to non-zero status.')
+                exit(1)
+            return True
+    return False
+
+
+def check_up(Debug=None, Host=None, domain='https://meta.miraheze.org', verify=True, force=False):
+    if not Debug and not Host:
+        raise Exception('Host or Debug must be specified')
+    if Debug:
+        server = f'{Debug}.miraheze.org'
+        headers = {'X-Miraheze-Debug': server}
+        location = f'{domain}@{server}'
+    else:
+        os.environ['NO_PROXY'] = 'localhost'
+        domain = 'https://localhost'
+        headers = {'host': Host}
+        location = f'{Host}@{domain}'
     up = False
-    headers = {'X-Miraheze-Debug': f'{server}.miraheze.org'}
-    req = requests.get('https://meta.miraheze.org/w/api.php?action=query&meta=siteinfo&formatversion=2&format=json', headers=headers)
-    if req.status_code == 200 and 'miraheze' in req.text and server in req.headers['X-Served-By']:
+    print(headers)
+    print(f'{domain}/w/api.php?action=query&meta=siteinfo&formatversion=2&format=json')
+    req = requests.get(f'{domain}/w/api.php?action=query&meta=siteinfo&formatversion=2&format=json', headers=headers, verify=verify)
+    if req.status_code == 200 and 'miraheze' in req.text and (Debug is None or Debug in req.headers['X-Served-By']):
         up = True
+    if not up:
+        print(f'Status: {req.status_code}')
+        print(f'Text: {"miraheze" in req.text}')
+        print(f'Debug: {(Debug is None or Debug in req.headers["X-Served-By"])}')
+        if force:
+            print(f'Ignoring canary check error on {location} due to --force')
+        else:
+            print(f'Canary check failed for {location}. Aborting... - use --force to proceed')
+            os.system(f'/usr/local/bin/logsalmsg DEPLOY ABORTED: Canary check failed for {location}')
```

Comment:
This should probably honour `--no-log`?

Reply:
Not a regression but fine to do

Reply:
Yeah I know just noting it during rewrite.